synapse/lib/storm.py

```diff
@@ -1469,3 +1470,302 @@ def getArgParser(self):
+                await node.set(name, oldv)
+            else:
+                runt.reqLayerAllowed(('prop:del', prop.full))
+                await node.pop(name)
+
+    async def undoPropDel(self, runt, splice, node):
+
+        name = splice.props.get('prop')
+        if name == '.created':
+            return
+
+        if node:
+            prop = node.form.props.get(name)
+            if prop is None:
+                raise s_exc.NoSuchProp(name=name, form=node.form.name)
+
+            valu = splice.props.get('valu')
+
+            runt.reqLayerAllowed(('prop:set', prop.full))
+            await node.set(name, valu)
+
+    async def undoNodeAdd(self, runt, splice, node):
+
+        if node:
+            for tag in node.tags.keys():
+                runt.reqLayerAllowed(('tag:del', *tag.split('.')))
+
+            runt.reqLayerAllowed(('node:del', node.form.name))
+            await node.delete(force=self.opts.force)
+
+    async def undoNodeDel(self, runt, splice, node):
+
+        if node is None:
+            form = splice.props.get('form')
+            valu = splice.props.get('valu')
+
+            if form and (valu is not None):
+                runt.reqLayerAllowed(('node:add', form))
+                await runt.snap.addNode(form, valu)
+
+    async def undoTagAdd(self, runt, splice, node):
+
+        if node:
+            tag = splice.props.get('tag')
+            parts = tag.split('.')
+            runt.reqLayerAllowed(('tag:del', *parts))
+
+            await node.delTag(tag)
+
+    async def undoTagDel(self, runt, splice, node):
+
+        if node:
+            tag = splice.props.get('tag')
+            parts = tag.split('.')
+            runt.reqLayerAllowed(('tag:add', *parts))
+
+            valu = splice.props.get('valu')
+            if valu is not None:
+                await node.addTag(tag, valu=valu)
+
+    async def undoTagPropSet(self, runt, splice, node):
+
+        if node:
+            tag = splice.props.get('tag')
+            parts = tag.split('.')
+            runt.reqLayerAllowed(('tag:del', *parts))
+
+            prop = splice.props.get('prop')
+
+            oldv = splice.props.get('oldv')
+            if oldv is not None:
+                runt.reqLayerAllowed(('tag:add', *parts))
+                await node.setTagProp(tag, prop, oldv)
+            else:
+                runt.reqLayerAllowed(('tag:del', *parts))
+                await node.delTagProp(tag, prop)
+
+    async def undoTagPropDel(self, runt, splice, node):
+
+        if node:
+            tag = splice.props.get('tag')
+            parts = tag.split('.')
+            runt.reqLayerAllowed(('tag:add', *parts))
+
+            prop = splice.props.get('prop')
+
+            valu = splice.props.get('valu')
+            if valu is not None:
+                await node.setTagProp(tag, prop, valu)
+
+    async def execStormCmd(self, runt, genr):
+
+        if self.opts.force:
+            if runt.user is not None and not runt.user.admin:
```

Comment:
I don't think we have scenarios where runt.user would be None anymore.  I'd be wiling to let this be a direct runt.user.admin check and allowing a None to blow up and let us know a None user got passed around; since a None user is a very bad problem.